### DISCUSSION TITLE
Recommended npm packaging flow

### DISCUSSION THREAD

**Discussion by Kuo-TingKai** [zhaohappy/libmedia#25] [Q&A]
Dear maintainer,

I would like to ask the recommended way to modify the configs for the npm packing flow for this scenario:
1. Re-compile the source code (`src/`), the new code includes new-added ts, .wgsl files and generate `dist/` , the entry point of package modules would be `index.js` (or `avplayer.js` in this scenario we only need avplayer module rather than avtranscoder etc.).
2. Only `npm publish` files in `dist/` into npm package registry for the user, they should not see the source code
3. User only need `npm install libmedia` for installation then he/she can `import {xxx} from libmedia` rather than import thins from node_modules/

Thanks

**Comment by zhaohappy** [comment]
There are several difficulties in publishing libmedia to npm at present:

1. Wasm file's dependency issues; unlike js, wasm files cannot packing into user code through npm package; users need to handle wasm's dependency and hosting wasm files themselves;

2. Due to the cheap's feature, users who want to  use the code below libmedia now can only depend on the source code and handle compilation issues themselves. In the future, the compiled js files may be publish to npm, but the toolchain has not yet been readyed;

3. avplayer and avtranscoder are compiled with dynamic modules. While these modules are publish to npm, users who import them into their own projects are likely to have problems;

Therefore, I don’t have any good suggestions on npm packing at present. The development way I recommend is to depend on the source code to develop and compile what you need.
